internal/tools/k8s/create.go

```diff
@@ -0,0 +1,268 @@
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+// CreateFromFile will create the Kubernetes resources in the provided file.
+//
+// The current behavior leads to the create being attempted on all resources in
+// the provided file. An error is returned if any of the create actions failed.
+// This process equates to running `kubectl create -f FILENAME`.
+func (kc *KubeClient) CreateFromFile(file ManifestFile) error {
+	data, err := ioutil.ReadFile(file.FQN())
+	if err != nil {
+		return err
+	}
+
+	apixv1beta1scheme.AddToScheme(scheme.Scheme)
+	mattermostscheme.AddToScheme(scheme.Scheme)
+
+	logger := kc.logger.WithFields(log.Fields{
+		"file": file.Name,
+	})
+
+	var failures int
+	resources := strings.Split(string(data), "---")
+	for _, resource := range resources {
+		if resource == "" {
+			continue
+		}
+		decode := scheme.Codecs.UniversalDeserializer().Decode
+
+		obj, _, err := decode([]byte(resource), nil, nil)
+		if err != nil {
+			logger.Error(errors.Wrap(err, "unable to decode k8s resource"))
+			failures++
+			continue
+		}
+
+		result, err := kc.createFileResource(file.DeployNamespace, obj)
+		if err != nil {
+			logger.Error(errors.Wrap(err, "unable to create k8s resource"))
+			failures++
+			continue
+		}
+
+		logger.Infof("Resource %q created!", result.GetName())
+	}
+
+	if failures > 0 {
+		return fmt.Errorf("encountered %d create failures", failures)
+	}
+
+	return nil
+}
+
+func (kc *KubeClient) createFileResource(deployNamespace string, obj interface{}) (metav1.Object, error) {
+	switch o := obj.(type) {
+	case *apiv1.ServiceAccount:
+		return kc.createServiceAccount(deployNamespace, obj.(*apiv1.ServiceAccount))
+	case *appsv1.Deployment:
+		return kc.createDeploymentV1(deployNamespace, obj.(*appsv1.Deployment))
+	case *appsbetav1.Deployment:
+		return kc.createDeploymentBetaV1(deployNamespace, obj.(*appsbetav1.Deployment))
+	case *rbacv1.RoleBinding:
+		return kc.createRoleBindingV1(deployNamespace, obj.(*rbacv1.RoleBinding))
+	case *rbacbetav1.RoleBinding:
+		return kc.createRoleBindingBetaV1(deployNamespace, obj.(*rbacbetav1.RoleBinding))
+	case *rbacv1.ClusterRole:
+		return kc.createClusterRoleV1(obj.(*rbacv1.ClusterRole))
+	case *rbacbetav1.ClusterRole:
+		return kc.createClusterRoleBetaV1(obj.(*rbacbetav1.ClusterRole))
+	case *rbacv1.ClusterRoleBinding:
+		return kc.createClusterRoleBindingV1(obj.(*rbacv1.ClusterRoleBinding))
+	case *rbacbetav1.ClusterRoleBinding:
+		return kc.createClusterRoleBindingBetaV1(obj.(*rbacbetav1.ClusterRoleBinding))
+	case *apixv1beta1.CustomResourceDefinition:
+		return kc.createCustomResourceDefinition(obj.(*apixv1beta1.CustomResourceDefinition))
+	case *mmv1alpha1.ClusterInstallation:
+		return kc.createClusterInstallation(deployNamespace, obj.(*mmv1alpha1.ClusterInstallation))
+	default:
+		return nil, fmt.Errorf("Error: unsupported k8s manifest type %T", o)
+	}
+}
+
+func (kc *KubeClient) createDeploymentV1(namespace string, deployment *appsv1.Deployment) (metav1.Object, error) {
```

Comment:
Yeah, it definitely isn't super streamlined. The complication comes from the fact that many operators like `mysql` have a standard deployment process that involves the provided mysql kubernetes yaml manifests.

Even though it may not be the most efficient process, I don't know that we want to deviate from it until we have a bit of experience under our belt running deployments.